### ISSUE TITLE
Answering any question

### ISSUE THREAD

**Issue by ritwik12** [ritwik12/Virtual-Assistant#32]
Here in this feature, the virtual assistant should be able to answer almost all the questions which are available on the internet. The basic approach is to fetch the answer(knowledge) from the internet (such as Google or Wikipedia) and passing it back to the answer.
The approach which I know is to get the data from Wikipedia using Wikipedia APIs and getting the data in JSON format and converting it to string for printing on the terminal. 
For further knowledge please see these links- 
https://stackoverflow.com/questions/8555320/is-there-a-clean-wikipedia-api-just-for-retrieve-content-summary

The JSON data will look like these links - 
https://en.wikipedia.org/w/api.php?format=json&action=query&prop=extracts&exintro=&explaintext=&titles=google
https://en.wikipedia.org/w/api.php?action=opensearch&search=facebook

The data can be fetched using above link and for converting it from JSON to string please refer our Restaurant feature.

**Comment by sanket-mali** [comment]
assign me on this..

**Comment by ritwik12** [comment]
@sanket-mali When this funtion will be invoked then we need to create another class/category for this functionality just like other functionalities (weather, restaurant, media, etc).

**Comment by ritwik12** [comment]
@sanket-mali Please forward with your work on this and if you face any problems I am here to help.

**Comment by sanket-mali** [comment]
In this functionality what will be the difference with the normal scenario like if we write any random word as of now it opens a browser window and search it with google..

**Comment by ritwik12** [comment]
@ritwik12 Right now we are just doing google search for the sentences we don't know. With this new functionality we can get the answer of questions just like Cortana or Siri or Google assistant answers when we ask something like "what is github?" or " which is the closest star?" etc

**Comment by vnksnkr** [comment]
Is this issue still open?I would like to work on it

**Comment by ritwik12** [comment]
@vnksnkr Yes this is open, please go ahead!!

**Comment by vnksnkr** [comment]
@ritwik12 thanks, I'm working on the wikipedia api..can you clarify on how to implement the same via google search?

**Comment by ritwik12** [comment]
@vnksnkr Fetching details directly from Wikipedia would not be an intelligent way to do this. It could be a part of the whole process but we need something better to intelligently scrap the internet and answer the question.

**Comment by vnksnkr** [comment]
@ritwik12 does google allow to scrape their search results? one way is to use their api

**Comment by ritwik12** [comment]
@vnksnkr No, it isn't too direct. We need an Algorithm here to work this out. Google API could be a part of it too but again not just fetching from that.

**Comment by vnksnkr** [comment]
@ritwik12 I'm sorry I couldn't get that..Cant we use a websearch api to which we can directly ask the questions and since search engines algorithms  like google always show the most relevant answer (highlighted in big fonts) as the first search result..just scrape the the top one?If that wasnt the feature we were looking could you provide an example on how the algorithm should respond?

**Comment by ritwik12** [comment]
@ritwik12 This is a Hack that we can use if we have the access to the Google Search results. We can look for a way to get the One Word or One Para answer from Google. For example:

`when is christmas?`
Google answers it simply like `Friday, 25 December`

But for cases like `What is Deep Learning?`

It does not give us a direct answer, in that case we need to fetch a brief summary from the first link or what google shows at right side of the page which is Wikipedia Definition as shown below:
<img width="1407" alt="Screenshot 2020-09-17 at 12 44 52 PM" src="https://user-images.githubusercontent.com/20038775/93520619-9c31de80-f8e3-11ea-82f2-df9a38d411f2.png">

Similarly there could be many cases. Like things like a simple science question etc, etc.

The goal of this project is to have learned by solving complex problems in unique ways. If it was a market product I would have suggested you to simply scrap from APIs and do the stuff.

**Comment by vnksnkr** [comment]
@ritwik12 Thanks for the clarification..my initial thought process was to collect relevant results using common sources like Wikipedia, Quora, google search etc and crank up some way to display the most relevant result

**Comment by vnksnkr** [comment]
@ritwik12 I was really looking forward to making a PR..unfortunately bringing up an ML algorithm is something out of my reach right now..although i would to learn it someday soon..still I'm really interested in this repo and I'm hoping to contribute..I'll probably look into some other issue and work my way around it. :)

**Comment by ritwik12** [comment]
@vnksnkr Thanks :) 
Sure, feel free to pick up any issue or if you think there is anything that will be good for this project. Feel free to open up new issues too. it could be a Bug fix, Feature request etc
